src/Native/System.Native/pal_readdir.cpp

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include "../config.h"
+#include "pal_readdir.h"
+#include <errno.h>
+#include <string.h>
+
+static void ConvertDirentToDirectoryEntry(dirent const &entry, DirectoryEntry* pOutput)
```

Comment:
Nit 1: const dirent& not dirent const &
Nit 2: let's not use hungarian notation, drop the 'p' prefix
Nit 3: I'd shorten this to just ConvertDirent or ConvertDirectoryEntry